app/pages/sections/supplier/supplier/controller.test.js

```diff
@@ -39,4 +39,64 @@ describe('supplier controller', () => {
       expect(contextCreator.getContext).toHaveBeenCalledWith({ orderId: 'order-1', supplierData: {} });
     });
   });
+
+  describe('putSupplier', () => {
+    afterEach(() => {
+      putData.mockReset();
+    });
+
+    const mockFormData = {
+      name: 'SupplierOne',
+      line1: 'line 1',
+      line2: '   line 2  ',
+      line3: '  line 3',
+      line4: null,
+      line5: 'line 5  ',
+      town: ' townville  ',
+      postcode: 'HA3 PSH',
+      firstName: 'Bob',
+    };
+
+    const formattedPutData = {
+      supplier: {
+        name: 'SupplierOne',
+        address: {
+          line1: 'line 1',
+          line2: 'line 2',
+          line3: 'line 3',
+          line5: 'line 5',
+          town: 'townville',
+          postcode: 'HA3 PSH',
+        },
+        primaryContact: {
+          firstName: 'Bob',
+        },
+      },
+    };
+
+    it('should format form data and call putData once with the correct params', async () => {
+      putData.mockResolvedValueOnce({});
+
+      await putSupplier({
+        orderId: 'order-id', data: mockFormData, accessToken: 'access_token',
+      });
+
+      expect(putData.mock.calls.length).toEqual(1);
+      expect(putData).toHaveBeenCalledWith({
+        endpoint: `${orderApiUrl}/api/v1/orders/order-id/sections/supplier`,
+        body: formattedPutData,
+        accessToken: 'access_token',
+        logger,
+      });
+    });
+
+    it('should return succes: true if put is successful', async () => {
```

Comment:
need tests for 400 or non-400

Reply:
I'm doing that as part of the next task. My bad I took bit of a shortcut and copy and pasted the implementation anyway.

Reply:
tut tut very bad 😄

Reply:
I know I feel dirty 😞